### PR TITLE
Refactor Cesium3DTilesStyle so that its always synchronous, add fromUrl

### PR DESCRIPTION
**Pull request by ggetz**
Fixes https://github.com/CesiumGS/cesium/issues/10345

This refactors `Cesium3DTilesStyle` so we can treat the object as completely synchronous, and removes the need for `ready` and `readyPromise` properties, which become deprecated. It also removes the `string` and `Resource` types from the constructor, again, deprecating them. Instead, this adds a `fromUrl` static function to asynchronously create a new style from a url.

This also allows us to clean up many of our specs, as we no longer need to wait on the style's `readyPromise` to be fulfilled.

TODO:
- [x] Create `remove in` issue

### CODE REVIEW

**Comment by ggetz** [comment]
@ebogo1 Could you please review?

**Review by ebogo1 — COMMENTED** [review]
Looks good! Just a few minor things. Tests well and I double checked Sandcastles for any deprecation warnings just in case.

File: Source/Scene/Cesium3DTileStyle.js [hunk 1]
```diff
@@ -223,9 +236,16 @@ Object.defineProperties(Cesium3DTileStyle.prototype, {
...
    * @type {Promise.<Cesium3DTileStyle>}
    * @readonly
+   * @deprecated
    */
   readyPromise: {
     get: function () {
+      //>>includeStart('debug', pragmas.debug);
+      deprecationWarning(
+        "readyPromise",
+        "readyPromise was deprecated in Cesium 1.94.  It will be removed in 1.96.  Use Cesium3DTileStyle.fromUrl instead if loading a style from a url."
```
[the remark this answers is not in the document]
Comment (ggetz) [reply Source/Scene/Cesium3DTileStyle.js]: That's supposed to be "if", but I reworded this sentence throughout to be clearer. Now it's "If loading a style from a url, use `Cesium3DTileStyle.fromUrl` instead."

**Comment by ebogo1** [comment]
Thanks @ggetz - just a typo in CHANGES.md and a small conflict with `main` but otherwise I think this is good to go :+1:

**Comment by ggetz** [comment]
Thanks again @ebogo1, fixed!

**Pull request merged**